### PR TITLE
daemon: Use DeepCopy method from StatusResponse

### PR DESCRIPTION
**Pull request by vadorovsky**
This change replaces usage of mohae/deepcopy with a generated DeepCopy
method of StatusResponse. mohae/deepcopy did not handle copying
strfmt.DateTime value properly.

Fixes: a9e759f ("daemon: do deep copy of model.StatusResponse")
Fixes: #6605

Signed-off-by: Michal Rostecki <mrostecki@suse.de>

---
This change is [Reviewable](https://reviewable.io/reviews/cilium/cilium/6650)

### CODE REVIEW

**Comment by vadorovsky** [comment]
test-me-please

File: vendor/github.com/go-openapi/strfmt/zz_generated.deepcopy.go [hunk 1]
```diff
@@ -0,0 +1,35 @@
+// +build !ignore_autogenerated
```
Comment (vadorovsky) [inline vendor/github.com/go-openapi/strfmt/zz_generated.deepcopy.go]: I'm open for suggestions where should I add this file. Or for general suggestions how we should add DeepCopy method to strfmt.

I was thinking about having a fork of strfmt, but that might be too much effort for just one file.

**Review by ianvernon — APPROVED** [review]

**Review by aanm — CHANGES_REQUESTED** [review]
When I first looked at this PR I got curious how you did solve the DeepCopy for DateTime, now I know

@mrostecki the reason I'm not ok with this change is because that we are modifying the vendor directory. All code in the vendor directory should be the same of the commit specified for that vendor in `Gopkg.toml`. There's no easy way to solve this problem AFAIK.

File: vendor/github.com/go-openapi/strfmt/zz_generated.deepcopy.go (hunk 1 above)
Comment (tgraf) [reply vendor/github.com/go-openapi/strfmt/zz_generated.deepcopy.go]: Can we submit it upstream?

Comment (vadorovsky) [reply vendor/github.com/go-openapi/strfmt/zz_generated.deepcopy.go]: Good idea, done https://github.com/go-openapi/strfmt/pull/44

**Comment by vadorovsky** [comment]
Let's wait for feedback on https://github.com/go-openapi/strfmt/pull/44

File: vendor/github.com/go-openapi/strfmt/zz_generated.deepcopy.go (hunk 1 above)
Comment (vadorovsky) [reply vendor/github.com/go-openapi/strfmt/zz_generated.deepcopy.go]: Upstream PR merged. I'm going to bump go-openapi version and update the PR.

Comment (tgraf) [reply vendor/github.com/go-openapi/strfmt/zz_generated.deepcopy.go]: Awesome!

**Comment by vadorovsky** [comment]
test-me-please

**Comment by vadorovsky** [comment]
test-me-please

**Comment by vadorovsky** [comment]
@aanm PTAL

**Review by aanm — APPROVED** [review]

**Pull request merged**